examples/lennard_jones_optimization.py

```diff
@@ -0,0 +1,116 @@
+import optax
+import numpy as np
+import jax.numpy as jnp
+import matplotlib.pyplot as plt
+import jax
+import tensorcircuit as tc
+
+
+jax.config.update("jax_enable_x64", True)
+K = tc.set_backend("jax")
+
+
+def calculate_potential(log_a, base_distance_matrix, epsilon=0.5, sigma=1.0):
+    """
+    Calculate the total Lennard-Jones potential energy for a given logarithm of the lattice constant (log_a).
+    """
+    lattice_constant = jnp.exp(log_a)
```

Comment:
better illustrate the example with lattice generated inside the function, otherwise it seems a static lattice can also do the task as long as the `base_distance_matrix` is given? In more complex case, when there are more lattice parameters, the "simply scaling the original distance matrix" paradigm will fail.